src/main/groovy/com/lesfurets/jenkins/unit/BasePipelineTest.groovy

```diff
@@ -25,13 +25,30 @@ abstract class BasePipelineTest {
 
     def stringInterceptor = { m -> m.variable }
 
+    def usernamePasswordInterceptor = { m -> [m.usernameVariable, m.passwordVariable] }
+
     def withCredentialsInterceptor = { list, closure ->
-        list.forEach {
-            binding.setVariable(it, "$it")
+        list.forEach { creds ->
+            if (creds instanceof String) {
+                binding.setVariable(creds, creds)                
+            } else {
+                creds.each {
+                    def var = it.toString()
```

Comment:
You can remove `if (creds instanceof String) {` and keep just  `def var = it.toString()` 
java has `String.toString()` so groovy should have it too.

Reply:
That is not the reason for it. I tried to avoid regression in the call stacks. Existing call stacks of the previous `string` method as well as one of your unit tests reusing the stringInterceptor for `file` method would change if I used a common list return for both cases.

The returned values from the two interceptors are not the same. One returns a single value as string, the other returns a list of string which is why creds.each is involved. I need to know if the value is a string or a list. The it.toString is probably overly cautious handling of the value which is expected to be a String or GString for the credential environment key provided by usernameVariable and passwordVariable.